39: Add guardian token to authentication

mbramson opened the pull request:
Adds token-based authentication.

Simplifies the Authenticate Plug because the User is retrieved from the database directly by Guardian whenever authentication occurs. This speeds up tests for authenticated routes slightly.

Closes #39

mbramson committed:
39: Add guardian token to authentication

ericworkman committed:
Add Guardian secret for prod

ericworkman commented:
@mbramson I've added the appropriate config to Heroku for this. The key was simply generated from `mix phoenix.gen.secret`. We can spend the time to generate a better key using params or pem files, but this should be fine. I don't think there's any need to add the generation method to the readme, unless you really want to.

mbramson commented:
I don't think that there is any need either. LGTM.

mbramson committed:
Refactor Authenticate Plug

Since the Guardian.Plug.current_resource grabs the User from Repo on its
own, we no longer need to retrieve the User within authenticate. This
does not affect production code, but makes the tests slightly quicker,
as they no longer need to do an extra Repo retrieval.
